b2sdk/b2http.py

```diff
@@ -375,6 +376,8 @@ def do_get():
             response = self.session.get(
                 url, headers=request_headers, stream=True, timeout=self.TIMEOUT
             )
+            # cast response to B2Response so we can override iter_content()
+            response.__class__ = B2Response
```

Comment:
I believe the right way to return a custom `Response` object is to subclass and install `HTTPAdapter` with a custom `build_response()`, but I cannot see a direct downside of using this approach.